docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
+authentication should "just work"
+as credentials can be loaded implicitly
+based on the environment you're application is running in
+and therefore there is no extra code required to authenticate.
+
+When running your app in production,
+you should rely on a service account.
+If you're running your application inside Google Compute Engine or
+Google App Engine,
+the service account can be automatically discovered.
+If you're running your application elsewhere,
+you should download the service account JSON keyfile
+and point to it using an environment variable.
+
+-------------------
+Order of evaluation
+-------------------
+
+``gcloud`` will look in a bunch of different places
```

Comment:
I think the order is wrong according to:
https://developers.google.com/identity/protocols/application-default-credentials
env -> gcloud -> app engine -> compute engine

or do you do something to change the order?